include/pybind11/detail/class.h

```diff
@@ -558,9 +533,12 @@ inline void enable_dynamic_attributes(PyHeapTypeObject *heap_type) {
     type->tp_traverse = pybind11_traverse;
     type->tp_clear = pybind11_clear;
 
-    static PyGetSetDef getset[] = {
-        {const_cast<char *>("__dict__"), pybind11_get_dict, pybind11_set_dict, nullptr, nullptr},
-        {nullptr, nullptr, nullptr, nullptr, nullptr}};
+    static PyGetSetDef getset[] = {{const_cast<char *>("__dict__"),
```

Comment:
When we drop 3.6 the `const_cast` can be removed (according to https://docs.python.org/3/c-api/structures.html#c.PyGetSetDef 3.7 is the first version where `name` is `const char *`).

While at it, could you please make the `const_cast` conditional on `PY_VERSION_HEX`?

Maybe (untested):
```
#if PY_VERSION_HEX <= 0x03060000
const_cast<char *>("__dict__"),
#else
"__dict__",
#endif
```

Then we will not overlook this when we drop 3.6.